quinn-proto/src/connection/mod.rs

```diff
@@ -393,11 +393,12 @@ where
     /// - the application performed some I/O on the connection
     /// - a call was made to `handle_event`
     /// - a call was made to `handle_timeout`
+    ///
+    /// `max_datagrams` specifies how many datagrams can be returned inside a
+    /// single Transmit using GSO. This must be at least 1.
     #[must_use]
-    pub fn poll_transmit(&mut self, now: Instant) -> Option<Transmit> {
-        // This will become a parameter to `poll_transmit` in an additional update
-        const MAX_DATAGRAMS: usize = 1;
-
+    pub fn poll_transmit(&mut self, now: Instant, max_datagrams: usize) -> Option<Transmit> {
```

Comment:
This is fine for use through the quinn API, but it seems unergonomic for applications that want to rely on quinn-proto directly. Since it looks like `max_datagrams` is constant over the lifetime of the `Connection`, should we stick in the `Connection` state somehow?

Reply:
I initially thought about adding it into `TransportConfig`. But the issue here is that `TransportConfig::default` can't set it, because it doesn't have access to platform specific data (defined in `quinn`). We could add another field, and make sure another constructor parameter is set when the quinn connection builds the proto connection. But that might actually require adding the parameter in a lot more places (since constructors delegate to each other).

Reply:
It looks like quinn-proto `Connection`s are only constructed in one place in quinn-proto `Endpoint`, and quinn-proto `Endpoint`s are only constructed in one place by the quinn `EndpointBuilder` (module some test code), so it doesn't seem that bad?

Reply:
I don't want this to show up in end user facing APIs like the existing `...Config` structs in quinn. Maybe we could introduce a `BackendConfig` that gets passed to the endpoint at the -proto layer, and then from the endpoint to the connections? That would be semantically fine, though significantly more code than the current approach, which I'm also fine with.

Reply:
Okay, let's keep this for now.